src/test/components/MenuButtons.test.js

```diff
@@ -14,56 +14,210 @@ import {
+    const getShareWithUrlsButton = () => getByValue('Share with URLs');
+    const getInnerShareWithUrlsButton = (): null | HTMLElement => {
+      const shareWithUrlsButton = getShareWithUrlsButton();
+      if (
+        shareWithUrlsButton.parentElement !== undefined &&
+        shareWithUrlsButton.parentElement !== null &&
+        shareWithUrlsButton.parentElement.nextElementSibling !== undefined &&
+        shareWithUrlsButton.parentElement.nextElementSibling !== null
+      ) {
+        return shareWithUrlsButton.parentElement.nextElementSibling.getElementsByClassName(
+          'arrowPanelOkButton'
+        )[0];
+      }
+      return null;
+    };
+
+    const getMenuButtonsContainer = () =>
+      getByTestId('menuButtonsCompositeButtonContainer');
+    const getPermalinkButton = () => getByValue('Permalink');
+
     return {
       store,
-      container,
+      ...renderResult,
+      getShareButton,
+      getInnerShareButton,
+      getShareWithUrlsButton,
+      getInnerShareWithUrlsButton,
+      getMenuButtonsContainer,
+      getPermalinkButton,
     };
   }
 
-  // profile.meta.networkURLsRemoved flag is set to false as a default.
-  it('renders the MenuButtons buttons', () => {
-    const { store, container } = setup();
-    store.dispatch(startSymbolicating());
+  describe('share and save buttons', function() {
+    // Mock hash
+    const hash = 'c5e53f9ab6aecef926d4be68c84f2de550e2ac2f';
 
-    // MenuButtons is rendering a fragment with several children. We need to
-    // check all children to assess that the component renders properly.
-    expect(Array.from(container.children)).toMatchSnapshot();
+    beforeAll(function() {
+      (window: any).TextEncoder = TextEncoder;
+    });
 
-    store.dispatch(doneSymbolicating());
-    expect(Array.from(container.children)).toMatchSnapshot();
-  });
+    afterAll(async function() {
+      delete (window: any).TextEncoder;
+    });
 
-  it('renders the MenuButtons buttons with profile.meta.networkURLsRemoved set to true', () => {
-    const profile = processProfile(createGeckoProfileWithJsTimings());
-    profile.meta.networkURLsRemoved = true;
-    const { store, container } = setup(profile);
-    store.dispatch(startSymbolicating());
+    beforeEach(function() {
+      // Flow doesn't know uploadBinaryProfileData is a jest mock.
+      (uploadBinaryProfileData: any).mockImplementation(() =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know sha1 is a jest mock.
+      (sha1: any).mockImplementation((_data: Uint8Array) =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know shortenUrl is a jest mock.
+      (shortenUrl: any).mockImplementation(() =>
+        Promise.resolve('https://perf-html.io/')
+      );
+    });
 
-    expect(Array.from(container.children)).toMatchSnapshot();
+    // profile.meta.networkURLsRemoved flag is set to false as a default.
+    it('render correctly', () => {
```

Comment:
I think this test could be a bit more precise. Snapshot failures are hard to debug, so it would be nice to split it into two `it` blocks, and mention that it's a snapshot test.

```js
it('matches the snapshot when starting to symbolicate', () => {
  const { store, container } = setup();
  store.dispatch(startSymbolicating());

  // MenuButtons is rendering a fragment with several children. We need to
  // check all children to assess that the component renders properly.
  expect(Array.from(container.children)).toMatchSnapshot();
});

it('matches the snapshot when done symbolicating', () => {
  const { store, container } = setup();
  store.dispatch(startSymbolicating());
  store.dispatch(doneSymbolicating());
  expect(Array.from(container.children)).toMatchSnapshot();
});
```

Reply:
Looking at this again, I think this should be the only snapshot test. Essentially the snapshots are really useful for knowing if the general rendered markup changed for a component, but not good for testing specific behavior and component logic. It's nice to have general coverage over the shape of the markup, but it's nice to really limit how many snapshots we have for a given test.